src/main/java/calculator/StringCalculator.java

```diff
@@ -0,0 +1,44 @@
+package calculator;
+
+import java.util.Arrays;
+
+public class StringCalculator {
+    private final DelimiterParser delimiterParser;
+
+    public StringCalculator() {
+        delimiterParser = new DelimiterParser();
+    }
+
+    public int calculate(String input) {
+        if (input.isEmpty()) {
+            return 0;
+        }
+
+        String[] numbers = delimiterParser.parseNumbers(input);
+
+        return sum(numbers);
+    }
+
+    private int sum(String[] numbers) {
+        return Arrays.stream(numbers)
+                .map(n -> {
+                    validateNumber(n);
+                    return Integer.parseInt(n);
+                })
+                .reduce(0, Integer::sum);
+    }
+
+    private void validateNumber(String number) {
```

Comment:
```suggestion

    private void validateNumber(String rawNumber) {
```

아직 파싱되지 않아서 `number`라고 부르면 조금 위험할 수도 있을 것 같아요.
저는 구분해주기 위해서 `raw`라는 키워드를 붙이는 것을 선호합니다! 참고만 해 주세요 😄